apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -79,3 +217,292 @@ def _from_table(self, arrow_table: pa.Table) -> pa.SparseCOOTensor:
             ]
         ).T
         return pa.SparseCOOTensor.from_numpy(coo_data, coo_coords, shape=self.shape)
+
+
+SingleAxisTableIterReturn = Iterator[BlockwiseTableReadIterResult]
+
+
+def _single_axis_blockwise_table_iter(
+    array: "SparseNDArray",
+    sr: clib.SOMAArray,
+    coords: options.SparseNDCoords,
+    axis: int,
+    size: int,
+    reindex_disable: List[int],
+    eager: bool = True,
+    pool: Optional[concurrent.futures.Executor] = None,
+) -> SingleAxisTableIterReturn:
```

Comment:
This and the below seem to have pretty similar structure. Would it make sense to put that common structure into the BlockwiseReadIter classes?

Reply:
I'm not entirely clear on the suggestion. Are you suggesting that the BlockwiseReadIter hold onto the common params, I the generators (such as this function) simply accept a BlockwiseReadIter as an argument?

(that does seem cleaner - I just wanted to confirm the suggest intent)

Reply:
It seems like these could be a method on the respective BlockwiseReadIter subclasses, and the common parts within them (it looks like some of the internal functions are substantially similar?) could be pulled into the base class (to the extent possible).

Reply:
Oh, I see - you want to move all this into the classes.  Let me give that some thought - it certainly seems possible, although I'm not sure it helps or hurts comprehensibility

Reply:
OK, just did a major refactoring along these lines.  PTAL!